### PR TITLE
vim: add vim-closer

### PR DESCRIPTION
**Pull request by onsails**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/misc/vim-plugins/overrides.nix [hunk 1]
```diff
@@ -395,6 +395,15 @@ self: super: {
     passthru.python3Dependencies = ps: with ps; [ beancount ];
   });
 
+  vim-closer = super.vim-closer.overrideAttrs(old: {
+    patches = [
+      (substituteAll {
+        src = ./patches/vim-closer/fix-tag.patch;
```
Comment (SuperSandro2000) [inline pkgs/misc/vim-plugins/overrides.nix:401]: Can we use a substituteInPlace or fetch an upstream patch?

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108949` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>spacevim</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108949` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
9 packages built:
</summary>
<ul>
<li>spacevim</li>
<li>vimPlugins.deol-nvim</li>
<li>vimPlugins.nvcode-color-schemes-vim</li>
<li>vimPlugins.nvim-dap</li>
<li>vimPlugins.vim-closer</li>
<li>vimPlugins.vim-floaterm</li>
<li>vimPlugins.vim-lsp</li>
<li>vimPlugins.vimspector</li>
<li>vimPlugins.yats-vim</li>
</ul>
</details>

File: pkgs/misc/vim-plugins/overrides.nix (hunk 1 above)
Comment (jonringer) [reply pkgs/misc/vim-plugins/overrides.nix:401]: better yet, there's nothing nixos specific in this, please upstream this

Comment (onsails) [reply pkgs/misc/vim-plugins/overrides.nix]: thanks for feedback, fixed

File: pkgs/misc/vim-plugins/overrides.nix [hunk 2]
```diff
@@ -395,6 +395,16 @@ self: super: {
     passthru.python3Dependencies = ps: with ps; [ beancount ];
   });
 
+  vim-closer = super.vim-closer.overrideAttrs(old: {
+    dependencies = with super; [ vim-misc ];
+    patches = [
+      (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/misc/vim-plugins/overrides.nix:401]:
```suggestion
      # Fix duplicate tag in doc
      (fetchpatch {
```

Comment (onsails) [reply pkgs/misc/vim-plugins/overrides.nix:401]: done

**Comment by SuperSandro2000** [comment]
```
hash mismatch in fixed-output derivation '/nix/store/1jmj0qzwai9prjrm0hg6cvqad58caa65-a504be8c7050e41b7dfc50c2362948e2cf7c5422.patch':
  wanted: sha256:0ni9mg86fdpa8pk4x80m5k48bm1hwpazrg644239q28vwplk2qf9
  got:    sha256:065q30d913fm3pc7r5y53wmnb7q7bhv21qxavm65bkb91242d409
cannot build derivation '/nix/store/j0fh8ykvja3ik6cnf0q3n2wkszh3zwfb-vimplugin-vim-closer-2020-10-24.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/srr39xzn69w98zniwcam584h4ck1qzar-env.drv': 1 dependencies couldn't be built
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108949` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>vimPlugins.vim-closer: log was empty</li>
  </ul>
</details>
<details>
<summary>
8 packages built:
</summary>
<ul>
<li>spacevim</li>
<li>vimPlugins.deol-nvim</li>
<li>vimPlugins.nvcode-color-schemes-vim</li>
<li>vimPlugins.nvim-dap</li>
<li>vimPlugins.vim-floaterm</li>
<li>vimPlugins.vim-lsp</li>
<li>vimPlugins.vimspector</li>
<li>vimPlugins.yats-vim</li>
</ul>
</details>

**Comment by onsails** [comment]
> ```
> hash mismatch in fixed-output derivation '/nix/store/1jmj0qzwai9prjrm0hg6cvqad58caa65-a504be8c7050e41b7dfc50c2362948e2cf7c5422.patch':
>   wanted: sha256:0ni9mg86fdpa8pk4x80m5k48bm1hwpazrg644239q28vwplk2qf9
>   got:    sha256:065q30d913fm3pc7r5y53wmnb7q7bhv21qxavm65bkb91242d409
> cannot build derivation '/nix/store/j0fh8ykvja3ik6cnf0q3n2wkszh3zwfb-vimplugin-vim-closer-2020-10-24.drv': 1 dependencies couldn't be built
> cannot build derivation '/nix/store/srr39xzn69w98zniwcam584h4ck1qzar-env.drv': 1 dependencies couldn't be built
> ```

Sometimes I don't understand how to work with sha256 correctly.
I've obtained my hash via
```
❯ nix-prefetch-url https://github.com/rstacruz/vim-closer/commit/a504be8c7050e41b7dfc50c2362948e2cf7c5422.patch
path is '/nix/store/q69cj0n5l7j7lcqdcj024p9s39dvwm5i-a504be8c7050e41b7dfc50c2362948e2cf7c5422.patch'
0ni9mg86fdpa8pk4x80m5k48bm1hwpazrg644239q28vwplk2qf9
```

and then `nix-build -A vimPlugins.vim-closer .` executed correctly.

**Comment by SuperSandro2000** [comment]
> Sometimes I don't understand how to work with sha256 correctly.
> I've obtained my hash via
> 
> ```
> ❯ nix-prefetch-url https://github.com/rstacruz/vim-closer/commit/a504be8c7050e41b7dfc50c2362948e2cf7c5422.patch
> path is '/nix/store/q69cj0n5l7j7lcqdcj024p9s39dvwm5i-a504be8c7050e41b7dfc50c2362948e2cf7c5422.patch'
> 0ni9mg86fdpa8pk4x80m5k48bm1hwpazrg644239q28vwplk2qf9
> ```
> 
> put it into derivation definition and then `nix-build -A vimPlugins.vim-closer .` executed correctly.

That does not work because the patch is normalized and edited before being put in the nix store and because we have no input aware fetchers nothing fails on your end.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108949` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
9 packages built:
</summary>
<ul>
<li>spacevim</li>
<li>vimPlugins.deol-nvim</li>
<li>vimPlugins.nvcode-color-schemes-vim</li>
<li>vimPlugins.nvim-dap</li>
<li>vimPlugins.vim-closer</li>
<li>vimPlugins.vim-floaterm</li>
<li>vimPlugins.vim-lsp</li>
<li>vimPlugins.vimspector</li>
<li>vimPlugins.yats-vim</li>
</ul>
</details>
